src/utils/cpg.ts

```diff
@@ -46,12 +47,23 @@ export function createActivityDefinition(action: Action): ActivityDefinition {
     publisher: 'Logged in user',
     description: action.description,
     kind: kind,
-    productCodeableConcept:
-      action.resource.resourceType === 'MedicationRequest'
-        ? action.resource.medicationCodeableConcept
-        : action.resource.code
+    productCodeableConcept: { coding: [] } // intentionally empty object, see below
   };
 
+  switch (kind) {
+    case 'MedicationRequest':
+      activityDefinition.productCodeableConcept = action.resource.medicationCodeableConcept;
+      break;
+    case 'ServiceRequest':
+    case 'CarePlan':
+      delete activityDefinition.productCodeableConcept; // not allowed on these in cqf-ruler
+      activityDefinition.code = action.resource.code;
+      break;
+    default:
+      // do nothing
+      break;
+  }
```

Comment:
Is there a reason `productCodeableConcept` is left here. The `ActivityDefinition` type adds `productCodeableConcept` as a required field but since it isn't anymore the model can be updated then simplify this code here
```suggestion
  };

  switch (kind) {
    case 'MedicationRequest':
      activityDefinition.productCodeableConcept = action.resource.medicationCodeableConcept;
      break;
    case 'ServiceRequest':
    case 'CarePlan':
      activityDefinition.code = action.resource.code;
      break;
    default:
      // do nothing
      break;
  }
```